### PR TITLE
Feat/reset password forgot password

### PR DESCRIPTION
**Pull request by Ezeh20**
This PR holds 
- returning users auth
- forgot password feat
- update password feat

Screenshot (295)
Screenshot (289)
Screenshot (288)
Screenshot (287)
Screenshot (286)
Screenshot (285)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ import { useScrollToTop } from './ScrollToTop/ScrollToTop';
 import LoadingMessage from './components/LoadingMessage';
 import useDetectInternet from './hooks/useDetectInternet';
 import NoInternet from './components/NoInternet/NoInternet';
+import { Toaster } from "react-hot-toast";
```
Comment (edustipend-root) [inline src/App.js:16]: Why did we add a new package? Can you add a screenshot of this toast to the PR so I see what it looks like?

File: src/components/ApplicationSteps/Step5Application/Internals/EmailVerified/EmailVerified.jsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React, { useEffect } from 'react';
 import { useContext, useState } from 'react';
-import jwt_decode from 'jwt-decode';
+import { jwtDecode } from 'jwt-decode';
```
Comment (edustipend-root) [inline src/components/ApplicationSteps/Step5Application/Internals/EmailVerified/EmailVerified.jsx:3]: Let's be careful about updating our dependencies. I see you updated this from `3.1.2` to `4.0.0`.
Package updates like this should be communicated and discussed on the team.

File: src/context/isAuthenticated.js
Comment (edustipend-root) [inline src/context/isAuthenticated.js:1]: If this file is empty @Ezeh20, we should delete it.

File: src/pages/reset-password/ResetPassword.jsx [hunk 3]
```diff
@@ -1,42 +1,54 @@
...
+      const jwt = window.location.search.split('=');
+      setToken(jwt[1]);
     }
-  }, []);
+
+    //validate the token
+    if (token) {
+      const decode = jwtDecode(token);
+      if (!decode) {
+        toast.error('Invalid token');
```
Comment (edustipend-root) [inline src/pages/reset-password/ResetPassword.jsx:38]: Please change this message from `Invalid token` to `Oops, the link you entered is invalid`

File: src/pages/reset-password/ResetPassword.jsx [hunk 4]
```diff
@@ -1,42 +1,54 @@
...
+        setTimeout(() => {
+          nav('/forgot-password');
+        }, 1500);
+      } else if (decode?.exp < Date.now() / 1000) {
+        toast.error('expired token');
```
Comment (edustipend-root) [inline src/pages/reset-password/ResetPassword.jsx:43]: Please change this message from `expired token` to `Oops, the link you entered is expired.`

**Review by edustipend-root — CHANGES_REQUESTED** [review]

File: src/pages/reset-password/ResetPassword.jsx (hunk 3 above)
Comment (Ezeh20) [reply src/pages/reset-password/ResetPassword.jsx:38]: done

File: src/pages/reset-password/ResetPassword.jsx (hunk 4 above)
Comment (Ezeh20) [reply src/pages/reset-password/ResetPassword.jsx:43]: done

File: src/context/isAuthenticated.js
Comment (Ezeh20) [reply src/context/isAuthenticated.js:1]: forgot to delete it

File: src/components/ApplicationSteps/Step5Application/Internals/EmailVerified/EmailVerified.jsx (hunk 2 above)
Comment (Ezeh20) [reply src/components/ApplicationSteps/Step5Application/Internals/EmailVerified/EmailVerified.jsx:3]: for some reason 3.1.2 could not decode the token. Decoding a token was used in only two places before now by me so i quickly changed the version before anyone else could use it. Isiaq and josh are currently using it.

File: src/App.js (hunk 1 above)
Comment (Ezeh20) [reply src/App.js:16]: toast was added for little and quick feedbacks

Screenshot (295)
Screenshot (294)

**Comment by edustipend-root** [comment]
@Ezeh20, please look at the PR again and let's resolve the merge conflicts.

**Review by edustipend-root — APPROVED** [review]

**Pull request merged**